pr_agent/algo/ai_handler.py

```diff
@@ -99,28 +109,39 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
                 )
             if self.azure:
                 model = 'azure/' + model
+            system = get_settings().get("CONFIG.MODEL_SYSTEM_PREFIX", "") + system + \
+                     get_settings().get("CONFIG.MODEL_SYSTEM_SUFFIX", "")
+            user = get_settings().get("CONFIG.MODEL_USER_PREFIX", "") + user + \
+                   get_settings().get("CONFIG.MODEL_USER_SUFFIX", "")
             messages = [{"role": "system", "content": system}, {"role": "user", "content": user}]
             response = await acompletion(
                 model=model,
                 deployment_id=deployment_id,
                 messages=messages,
                 temperature=temperature,
-                force_timeout=get_settings().config.ai_timeout
+                force_timeout=get_settings().config.ai_timeout,
+                api_base=self.api_base,
             )
-        except (APIError, Timeout, TryAgain) as e:
+        except (APIError, TryAgain) as e:
             get_logger().error("Error during OpenAI inference: ", e)
             raise
-        except (RateLimitError) as e:
+        except RateLimitError as e:
             get_logger().error("Rate limit error during OpenAI inference: ", e)
             raise
-        except (Exception) as e:
+        except Exception as e:
             get_logger().error("Unknown error during OpenAI inference: ", e)
             raise TryAgain from e
```

Comment:
**Suggestion:** Consider using a more specific exception instead of the generic `Exception` in the `chat_completion` method. This will make the error handling more precise and prevent catching and handling unexpected exceptions that might need different handling logic. [best practice]
```suggestion
        except SpecificException as e:  # Replace SpecificException with the specific exception you want to catch
            get_logger().error("Specific error during OpenAI inference: ", e)
            raise TryAgain from e
```